src/react/utils.js

```diff
@@ -60,23 +60,24 @@ export function isReactElement(val: Value): boolean {
   if (realm.react.reactElements.has(val)) {
     return true;
   }
-  if (val.properties.has("$$typeof")) {
-    let $$typeof = Get(realm, val, "$$typeof");
-    let globalObject = realm.$GlobalObject;
-    let globalSymbolValue = Get(realm, globalObject, "Symbol");
-
-    if (globalSymbolValue === realm.intrinsics.undefined) {
-      if ($$typeof instanceof NumberValue) {
-        return $$typeof.value === 0xeac7;
-      }
-    } else if ($$typeof instanceof SymbolValue) {
-      let symbolFromRegistry = realm.globalSymbolRegistry.find(e => e.$Symbol === $$typeof);
-      let _isReactElement = symbolFromRegistry !== undefined && symbolFromRegistry.$Key === "react.element";
-      if (_isReactElement) {
-        // add to Set to speed up future lookups
-        realm.react.reactElements.add(val);
-        return true;
-      }
+  if (!val.properties.has("type") || !val.properties.has("props") || !val.properties.has("$$typeof")) {
+    return false;
+  }
+  let $$typeof = Get(realm, val, "$$typeof");
```

Comment:
Calling Get here is not safe since it you may end up calling a getter that has side-effects. It is unlikely that the caller of isReactElement is going to always be careful to make sure that only side-effects that are expected by the application programmer survives.

There is a helper method: _SafeGetDataPropertyValue that you can use to get the value of a local (i.e. not from the prototype) data (i.e. no getter) property, in such a way that no invariant checking code is generated (i.e. no sneaky side effects).

Reply:
This isn't new code introduced in this PR, so I'm happy to fix in a follow up PR too.